API docs creation (#6)
kubecost/docs-cloud: apis/api-directory/cloud-cost-api.md

# Cloud Cost API

{% swagger method="get" path="query/cloudcost/query" baseUrl="http://app.kubecost.com/model" summary="Cloud Cost API (Querying)" %}
{% swagger-description %}
The Cloud Cost API is the preferred way to query for costs and resources related to cloud provider services.
{% endswagger-description %}

{% swagger-parameter in="path" name="window" required="true" type="string" %}
Duration of time over which to query. Accepts multiple formats including units of time, relative time units, and unix timestamps. See this section on [using the `window` parameter](/apis/api-directory/api-directory.md#using-the-window-parameter) for more information.

> {% endswagger-parameter %}

Author: Having some problems generating data when querying for roughly less than 12 hours. Returns a 200 but no data

Author: This seems to also apply to Allocation/Assets, sampling under 24h doesn't return any data

Reviewer: The resolution for cloud costs is 24h, but you should be able to get data for assets/allocations if querying under 24h. I'll take a look at this :)

{% swagger-parameter in="path" name="format" type="string" required="false" %}
File type when exporting query. Currently only supports `json`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="aggregate" type="string" required="false" %}

> Used to consolidate cost model data. Supported values are `account`, `cluster`, `project`, `providerid`, `provider`, and `type`. Passing an empty value for this parameter or none at all returns data by an individual asset. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate=account,project`.

Reviewer: ```suggestion
Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual asset. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate=account,project`.

```

{% endswagger-parameter %}

{% swagger-parameter in="path" name="accumulate" type="boolean" required="false" %}

> When set to `false`, this endpoint returns daily time series data vs cumulative data. Default value is `false`.

Reviewer: Should this match the description for `accumulate` in allocations? 
```suggestion
If `true`, sum the entire range of time intervals into a single set. Default value is `false`.

```

{% endswagger-parameter %}

{% swagger-parameter in="path" name="costMetric" required="false" %}
Determines which cloud cost metric type will be returned. Acceptable values are `amortizedNetCost`, `amortizedCost`, `invoicedCost`, `listCost`, and `netCost`. Default is `amortizedNetCost`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="limit" type="int" required="false" %}

> Refers to the number of line items per page. You should also set `accumulate=true` to obtain a single list of line items, otherwise you will receive a group of line items per interval of time being sampled. Paginates by all item types, and will provide lists for those types.

Reviewer: Same as for allocations :) 
```suggestion
Refers to the number of line items per page. Currently, only supported together with `accumulate=true` to obtain a single list of line items.
```

# Cloud Cost API

{% swagger method="get" path="/cloudcost/query" baseUrl="http://app.kubecost.com/query" summary="Cloud Cost API (Querying)" %}
{% swagger-description %}
The Cloud Cost API is the preferred way to query for costs and resources related to cloud provider services.
{% endswagger-description %}

{% swagger-parameter in="path" name="window" required="true" type="string" %}
Duration of time over which to query. Accepts multiple formats including units of time, relative time units, and unix timestamps. See this section on [using the `window` parameter](/apis/api-directory/api-directory.md#using-the-window-parameter) for more information.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="format" type="string" required="false" %}
File type when exporting query. Currently only supports `json`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="aggregate" type="string" required="false" %}

> Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual asset. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate=account,project`.

Reviewer: These are **all** the supported agg options for cloud. No further action required here.

> Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual asset. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate=account,project`.

Reviewer: ```suggestion
Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual cloud cost. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate= provider,service`.

```

{% endswagger-parameter %}

{% swagger-parameter in="path" name="accumulate" type="boolean" required="false" %}
If `true`, sum the entire range of time intervals into a single set. Default value is `false`.

{% endswagger-parameter %}

{% swagger-parameter in="path" name="costMetric" required="false" %}
Determines which cloud cost metric type will be returned. Acceptable values are `amortizedNetCost`, `amortizedCost`, `invoicedCost`, `listCost`, and `netCost`. Default is `amortizedNetCost`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="limit" type="int" required="false" %}
Refers to the number of line items per page. Currently, only supported together with `accumulate=true` to obtain a single list of line items.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="filter" type="string" required="false" %}

> Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by clusters A and B, use `filter=cluster:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

Reviewer: Same here:

```suggestion
Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

```

{% endswagger-parameter %}

{% swagger method="get" path="/cloudcost/totals" baseUrl="http://app.kubecost.com/query" summary="Cloud Costs API (Total)" %}
{% swagger-description %}

> The `/totals` endpoint does not aggregate, and accepts only a window and filter, as well as any optional filters, returning a single total cost.

Reviewer: ```suggestion
The `/totals` endpoint does not aggregate, and accepts only a window and an optional filter, returning a single total cost.

```

{% endswagger-description %}

{% swagger-parameter in="path" name="window" type="string" required="true" %}
Duration of time over which to query. Accepts multiple formats including units of time, relative time units, and unix timestamps. See this section on [using the `window` parameter](/apis/api-directory/api-directory.md#using-the-window-parameter) for more information.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="format" type="string" required="false" %}
File type when exporting query. Currently only supports `json`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="filter" type="string" required="false" %}

> Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by clusters A and B, use `filter=cluster:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

Reviewer: Cluster filters aren't supported for cloud costs, we should probably avoid confusion?

```suggestion
Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

```

# Cloud Cost API

{% swagger method="get" path="/cloudcost/query" baseUrl="http://app.kubecost.com/query" summary="Cloud Cost API (Querying)" %}
{% swagger-description %}
The Cloud Cost API is the preferred way to query for costs and resources related to cloud provider services.
{% endswagger-description %}

{% swagger-parameter in="path" name="window" required="true" type="string" %}
Duration of time over which to query. Accepts multiple formats including units of time, relative time units, and unix timestamps. See this section on [using the `window` parameter](/apis/api-directory/api-directory.md#using-the-window-parameter) for more information.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="format" type="string" required="false" %}
File type when exporting query. Currently only supports `json`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="aggregate" type="string" required="false" %}

> Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual cloud cost. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate= provider,service`.

Reviewer: ```suggestion
Used to consolidate cost model data. Supported values are `invoiceEntityID`, `accountID`, `provider`, `providerID`, `category`, and `service`, as well as `label:<name>`. Passing an empty value for this parameter or none at all returns data by an individual cloud cost. Supports multi-aggregation (aggregation of multiple categories) in a comma separated list, such as `aggregate=provider,service`.

```

{% endswagger-parameter %}

{% swagger-parameter in="path" name="accumulate" type="boolean" required="false" %}
If `true`, sum the entire range of time intervals into a single set. Default value is `false`.

{% endswagger-parameter %}

{% swagger-parameter in="path" name="costMetric" required="false" %}
Determines which cloud cost metric type will be returned. Acceptable values are `amortizedNetCost`, `amortizedCost`, `invoicedCost`, `listCost`, and `netCost`. Default is `amortizedNetCost`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="limit" type="int" required="false" %}
Refers to the number of line items per page. Currently, only supported together with `accumulate=true` to obtain a single list of line items.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="filter" type="string" required="false" %}

> Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

Reviewer: ```suggestion
Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:categoryA,categoryB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

```

{% endswagger-parameter %}

{% swagger method="get" path="/cloudcost/totals" baseUrl="http://app.kubecost.com/query" summary="Cloud Costs API (Total)" %}
{% swagger-description %}
The `/totals` endpoint does not aggregate, and accepts only a window and an optional filter, returning a single total cost.

{% endswagger-description %}

{% swagger-parameter in="path" name="window" type="string" required="true" %}
Duration of time over which to query. Accepts multiple formats including units of time, relative time units, and unix timestamps. See this section on [using the `window` parameter](/apis/api-directory/api-directory.md#using-the-window-parameter) for more information.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="format" type="string" required="false" %}
File type when exporting query. Currently only supports `json`.
{% endswagger-parameter %}

{% swagger-parameter in="path" name="filter" type="string" required="false" %}

> Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:clusterA,clusterB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

Reviewer: ```suggestion
Filter your results by any category which you can aggregate by, can support multiple filterable items in the same category in a comma-separated list. For example, to filter results by categories A and B, use `filter=category:categoryA,categoryB` See our [Filter Parameters](/apis/filter-parameters.md) doc for a complete explanation of how to use filters and what categories are supported.

```
